File: src/services/handlers/deleteKey.ts
import fs        from "node:fs";
import log       from "../../log.js";
import {Options} from "../../components/qualifyingRules.js";

export default function deleteKey(options: Options) {
  const key: string = options.targetKeys[0];
  log.debug(`Deleting "${key}"`);

  if (options.envObject[key]) {
    const lineStart = options.envObject[key].lineStart;
    const lineEnd   = options.envObject[key].lineEnd;
    log.debug(`Deleting lines ${lineStart}-${lineEnd}`);

    // Read the file and split it into an array of lines
    let lines: string[] = fs.readFileSync(options.fullEnvPath, 'utf8').split('\n');

    // Remove the lines between lineStart and lineEnd
    lines.splice(lineStart, lineEnd - lineStart + 1);

    // Join the lines back together and write the result back to the file
    fs.writeFileSync(options.fullEnvPath, lines.join('\n'));
  } else {
    log.debug(`Environment variable "${key}" not found`);
    process.exitCode = 1;
  }
};

File: src/handlers/delete_key.rs
use std::fs;

use crate::qualifying_rules::Options;

/// Remove a key and its value lines from the .env file.
///
/// Returns `true` if the key was found and deleted, `false` if not found
/// (caller should set exit code 1).
pub fn delete_key(options: &Options) -> bool {
    let key = &options.target_keys[0];
    let env_object = options.env_object.as_ref().unwrap();

    if let Some(env_val) = env_object.get(key) {
        let content = fs::read_to_string(&options.full_env_path).expect("Failed to read .env file");
        let mut lines: Vec<String> = content.split('\n').map(str::to_string).collect();

        let start = env_val.line_start as usize;
        let end = env_val.line_end as usize;
        lines.drain(start..=end);

        fs::write(&options.full_env_path, lines.join("\n")).expect("Failed to write .env file");
        true
    } else {
        false
    }
}
